stock_move_propagate_first_move_cancel/models/stock_move.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockMove(models.Model):
+
+    _inherit = "stock.move"
+
+    def _action_cancel(self):
+        res = super()._action_cancel()
+        moves_to_cancel = self.search(
```

Comment:
@sbejaoui Thinking on this, shouldn't it use the stock.rule 'propagate _cancel' attribute and add that in the domain?